azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_nuget.go

```diff
@@ -0,0 +1,181 @@
+		Schema: map[string]*schema.Schema{
+			"username": {
+				Description: "The Nuget feed user name.",
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+			},
+			"password": {
+				Description: "The Nuget feed password.",
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+			},
+		},
+	}
+
+	r.Schema["authentication_token"] = &schema.Schema{
+		Type:         schema.TypeList,
+		Optional:     true,
+		MinItems:     1,
+		MaxItems:     1,
+		Elem:         at,
+		ExactlyOneOf: []string{"authentication_basic", "authentication_token", "authentication_none"},
+	}
+
+	r.Schema["authentication_none"] = &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MinItems: 1,
+		MaxItems: 1,
+		Elem:     ak,
+	}
+
+	r.Schema["authentication_basic"] = &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MinItems: 1,
+		MaxItems: 1,
+		Elem:     aup,
+	}
+
+	return r
+}
+
+// Convert internal Terraform data structure to an AzDO data structure
+func expandServiceEndpointNuget(d *schema.ResourceData) (*serviceendpoint.ServiceEndpoint, *uuid.UUID, error) {
+	serviceEndpoint, projectID := doBaseExpansion(d)
+	serviceEndpoint.Type = converter.String("externalnugetfeed")
+	serviceEndpoint.Url = converter.String(d.Get("url").(string))
+	authScheme := "Token"
+
+	authParams := make(map[string]string)
+
+	if x, ok := d.GetOk("authentication_token"); ok {
+		authScheme = "Token"
+		msi := x.([]interface{})[0].(map[string]interface{})
+		authParams["apitoken"], ok = msi["token"].(string)
+		if !ok {
+			return nil, nil, errors.New("Unable to read 'token'")
+		}
+	} else if x, ok := d.GetOk("authentication_none"); ok {
+		authScheme = "None"
+		msi := x.([]interface{})[0].(map[string]interface{})
+		authParams["nugetkey"], ok = msi["key"].(string)
+		if !ok {
+			return nil, nil, errors.New("Unable to read 'key'")
+		}
+	} else if x, ok := d.GetOk("authentication_basic"); ok {
+		authScheme = "UsernamePassword"
+		msi := x.([]interface{})[0].(map[string]interface{})
+		authParams["username"], ok = msi["username"].(string)
+		if !ok {
+			return nil, nil, errors.New("Unable to read 'username'")
+		}
+		authParams["password"], ok = msi["password"].(string)
+		if !ok {
+			return nil, nil, errors.New("Unable to read 'password'")
+		}
+	}
+	serviceEndpoint.Authorization = &serviceendpoint.EndpointAuthorization{
+		Parameters: &authParams,
+		Scheme:     &authScheme,
+	}
+
+	return serviceEndpoint, projectID, nil
+}
+
+// Convert AzDO data structure to internal Terraform data structure
+func flattenServiceEndpointNuget(d *schema.ResourceData, serviceEndpoint *serviceendpoint.ServiceEndpoint, projectID *uuid.UUID) {
+	doBaseFlattening(d, serviceEndpoint, projectID)
+	if strings.EqualFold(*serviceEndpoint.Authorization.Scheme, "UsernamePassword") {
+		if _, ok := d.GetOk("authentication_basic"); !ok {
+			auth := make(map[string]interface{})
+			auth["username"] = ""
```

Comment:
oops, username should be read in this case. With the Artifactory ones it doesn't work because the username field got marked as sensitive in the AzDO service connection extension by accident. JFrog will fix that in their next extension release. For all practical purposes it doesn't matter, but may want to fix it for upstreaming. @innagarc